src/IO/H5/Header.hpp

```diff
@@ -0,0 +1,53 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines class h5::Header
+
+#pragma once
+
+#include <string>
+
+#include "IO/H5/Object.hpp"
+#include "IO/H5/OpenGroup.hpp"
+
+namespace h5 {
+/*!
+ * \ingroup HDF5
+ * \brief Writes header info about the build, git commit, branch, etc.
```

Comment:
@kidder any thoughts? This will depend on what the specific h5::Object most likely. E.g. appending a Dat file and changing the format would be a breaking change so appending would be a user error and would only work if the same number of columns are present.